src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/transport/OpenDistroSecuritySSLNettyTransport.java

```diff
@@ -178,21 +198,42 @@ public void connect(ChannelHandlerContext ctx, SocketAddress remoteAddress, Sock
     protected class SSLClientChannelInitializer extends Netty4Transport.ClientChannelInitializer {
         private final boolean hostnameVerificationEnabled;
         private final boolean hostnameVerificationResovleHostName;
+        private final DiscoveryNode node;
+        private SSLConnectionTestResult connectionTestResult;
 
         public SSLClientChannelInitializer(DiscoveryNode node) {
+            this.node = node;
             hostnameVerificationEnabled = settings.getAsBoolean(
                     SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION, true);
             hostnameVerificationResovleHostName = settings.getAsBoolean(
                     SSLConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION_RESOLVE_HOST_NAME, true);
+
+            connectionTestResult = SSLConnectionTestResult.SSL_AVAILABLE;
+            if (OpenDistroSSLDualModeConfig.getInstance().isDualModeEnabled()) {
+                SSLConnectionTestUtil sslConnectionTestUtil = new SSLConnectionTestUtil(node.getAddress().getAddress(), node.getAddress().getPort());
+                connectionTestResult = AccessController.doPrivileged((PrivilegedAction<SSLConnectionTestResult>) sslConnectionTestUtil::testConnection);
```

Comment:
if other node is only communicating in SSL mode (means other node has older version of the security plugin which is not supporting dual mode), connectionTestResult will be always ES_PING_FAILED  .. is that true? 


    public SSLConnectionTestResult testConnection() {
        if (sendDualSSLClientHello()) {.    --> This will fail as server doesn't send DualSSL Hello 
            return SSLConnectionTestResult.SSL_AVAILABLE;
        }

        if (sendESPing()) {.  ----> this will fail too as other node is actually sending SSL response...  
            return SSLConnectionTestResult.SSL_NOT_AVAILABLE;
        }

        return SSLConnectionTestResult.ES_PING_FAILED;   --> this will be return..
    }